app/code/Magento/Theme/view/frontend/web/js/view/messages.js

```diff
@@ -25,9 +25,17 @@ define([
          * Extends Component object by storage observable messages.
          */
         initialize: function () {
+            var SCRIPT_REGEX = /<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi;
+
             this._super();
 
             this.cookieMessages = _.unique($.cookieStorage.get('mage-messages'), 'text');
+            this.cookieMessages.forEach(function (value) {
+                value.text = _.unescape(value.text);
+                value.text = value.text.replace(SCRIPT_REGEX, "");
+
```

Comment:
@edenduong Instead of unescaping message and removing script tags, I think whitelisting only anchor tag with href attribute would be more painless and secure.

Reply:
@kalpmehta: I have solved it. Please check it again. Thanks!

Reply:
@edenduong Thanks! Do you think this can be supported with tests?

Reply:
@kalpmehta : I just cover it by MFTF Test. Please check it. Thanks!

Reply:
this has no place in core Magento because it's a really dirty hack. I don't see the commit where `a` tag is whitelisted instead

Reply:
@AlexMaxHorkun Thanks! Do you have suggestion to render only `a` tag and also ensure it's secure?